src/layout/page-layout.scss

```diff
@@ -0,0 +1,384 @@
+  // stylelint-disable length-zero-no-unit
+  --Layout-outer-spacing-x: 0px; // wrapper margin x
+  --Layout-outer-spacing-y: 0px; // wrapper margin y
+  --Layout-inner-spacing-min: 0px; // default region padding
+  --Layout-inner-spacing-max: 0px; // relaxed content horizontal padding
+  // stylelint-enable length-zero-no-unit
+}
+
+.PageLayout {
+  // stylelint-disable-next-line primer/spacing
+  margin: var(--Layout-outer-spacing-y) var(--Layout-outer-spacing-x);
+
+  > .PageLayout-wrapper {
+    display: grid;
+    grid: auto-flow / 1fr;
+    row-gap: var(--Layout-row-gap);
+
+    > .PageLayout-columns {
+      display: grid;
+      column-gap: var(--Layout-column-gap);
+      row-gap: var(--Layout-row-gap);
+      grid-template-columns: var(--Layout-template-columns);
+      grid-template-rows: 1fr;
+      grid-template-areas: var(--Layout-template-areas);
+    }
+  }
+
+  // outer spacing
+
+  &.PageLayout--outerSpacing-normal {
+    --Layout-outer-spacing-x: #{$spacer-3}; // 16px
+    --Layout-outer-spacing-y: #{$spacer-3}; // 16px
+
+    @include breakpoint(lg) {
+      --Layout-outer-spacing-x: #{$spacer-4}; // 24px
+      --Layout-outer-spacing-y: #{$spacer-4}; // 24px
+    }
+  }
+
+  &.PageLayout--outerSpacing-condensed {
+    --Layout-outer-spacing-x: #{$spacer-3}; // 16px
+    --Layout-outer-spacing-y: #{$spacer-3}; // 16px
+  }
+
+  // inner spacing
+
+  &.PageLayout--innerSpacing-normal {
+    --Layout-inner-spacing-min: #{$spacer-3}; // 16px
+    --Layout-inner-spacing-max: #{$spacer-3}; // 16px
+
+    @include breakpoint(lg) {
+      --Layout-inner-spacing-max: #{$spacer-4}; // 24px
+    }
+  }
+
+  &.PageLayout--innerSpacing-condensed {
+    --Layout-inner-spacing-min: #{$spacer-3}; // 16px
+    --Layout-inner-spacing-max: #{$spacer-3}; // 16px
+  }
+
+  // column gap
+
+  &.PageLayout--columnGap-normal {
+    --Layout-column-gap: #{$spacer-3}; // 16px
+
+    @include breakpoint(lg) {
+      --Layout-column-gap: #{$spacer-4}; // 24px
+    }
+  }
+
+  &.PageLayout--columnGap-condensed {
+    --Layout-column-gap: #{$spacer-3}; // 16px
+  }
+
+  &.PageLayout--columnGap-none {
+    // the `px` unit is mandatory for `calc()` execution. See https://stackoverflow.com/a/32518348
+    // stylelint-disable-next-line length-zero-no-unit
+    --Layout-column-gap: 0px;
+  }
+
+  // row gap
+
+  &.PageLayout--rowGap-normal {
+    --Layout-row-gap: #{$spacer-3}; // 16px
+
+    @include breakpoint(lg) {
+      --Layout-row-gap: #{$spacer-4}; // 24px
+    }
+  }
+
+  &.PageLayout--rowGap-none {
+    // the `px` unit is mandatory for `calc()` execution. See https://stackoverflow.com/a/32518348
+    // stylelint-disable length-zero-no-unit
+    --Layout-row-gap: 0px;
+  }
+
+  &.PageLayout--rowGap-condensed {
+    --Layout-row-gap: #{$spacer-3}; // 16px
+  }
+
+  // regions
+
+  .PageLayout-header,
+  .PageLayout-content,
+  .PageLayout-pane,
+  .PageLayout-footer {
```

Comment:
Are we sure these have to be descendent selectors?

Reply:
They don't :) Fixed it!